### PR TITLE
Improved return type of `Reflection*::getModifiers()`

### PR DESCRIPTION
**Pull request by kukulich**
Constant should be removed in https://github.com/Roave/BetterReflection/pull/1235/commits/a4e5618c5907c00e9315e8d3b3547d8c8679a4e9

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -25,3 +25,6 @@ parameters:
         -
             message: '#Call to an undefined method PhpParser\\NodeVisitorAbstract::(getNode|setConstantName)\(\)#'
             path: src/SourceLocator/Type/AutoloadSourceLocator.php
+        -
+            message: '#Method Roave\\BetterReflection\\Reflection\\ReflectionClass::computeModifiers\(\) never returns \d+ so it can be removed from the return type#'
```
Comment (Ocramius) [inline phpstan.neon:29]: There's a grammar error here on `returns` /cc @ondrejmirtes

File: src/Reflection/ReflectionClass.php [hunk 2]
```diff
@@ -332,6 +326,7 @@ private function createMethodsFromTrait(ReflectionMethod $method): array
 
         if (array_key_exists($methodHash, $this->traitsData['modifiers'])) {
             // PhpParser modifiers are compatible with PHP reflection modifiers
+            /** @var int-mask-of<ReflectionMethodAdapter::IS_*> $methodModifiers */
```
Comment (Ocramius) [inline src/Reflection/ReflectionClass.php]: Can this not be inferred? 🤔

**Review by Ocramius — APPROVED** [review]
👍

Even more type improvements, nice!

File: src/Reflection/ReflectionClass.php (hunk 2 above)
Comment (kukulich) [reply src/Reflection/ReflectionClass.php]: Cherry-picked two commits from my next PR.

Psalm can infer the type now, PHPStan cannot.
https://github.com/Roave/BetterReflection/pull/1236/commits/8f89aae63ac23aa1b5aa85d881db7f5b32fdf467

**Pull request merged**